### PR TITLE
Add provided certificate checking before LE certificate generation with OnHostRule option

### PR DESCRIPTION
**Pull request by nmengin**
During the code review of PR #1260, it appears that the provided certificates are not checked before LE certificate checking with the `OnHostRule` option.

Adding this check allows to generate a LE certificate only if domains have no provided certificates.

Thanks to it, PR #1260 will be modified to apply filter on domains to check by provided certificate in place of domains to ignore during LE certificate generation.

cc/ @CyrilPeponnet

### CODE REVIEW

**Comment by nmengin** [comment]
@emilevauge I am agree with you for the tests.
We discussed about it with @ldez .

We believe that it can be nice to upgrade the boulder image and create ACME tests into a new PR.
I'm going to propose a new PR about this subject ASAP.

File: integration/acme_test.go [hunk 1]
```diff
@@ -4,12 +4,14 @@ import (
 	"crypto/tls"
 	"net/http"
 	"os"
-	"os/exec"
 	"time"
 
 	"github.com/go-check/check"
 
 	"errors"
+
```
Comment (ldez) [inline integration/acme_test.go]: could you reformat imports?

File: acme/acme_test.go [hunk 2]
```diff
@@ -9,6 +9,9 @@ import (
 	"testing"
 	"time"
 
+	"crypto/tls"
```
Comment (ldez) [inline acme/acme_test.go]: could you reformat imports?

File: integration/acme_test.go [hunk 3]
```diff
@@ -3,18 +3,22 @@ package main
...
+	"github.com/containous/traefik/integration/utils"
+	"github.com/go-check/check"
+	checker "github.com/vdemeester/shakers"
 )
 
 // ACME test suites (using libcompose)
 type AcmeSuite struct {
 	BaseSuite
+	onDemand            bool
```
Comment (ldez) [inline integration/acme_test.go]: I think it's better to create a dedicated struct.

Comment (nmengin) [inline integration/acme_test.go]: The others tests use one struct for BaseSuite and all the arguments they need.
I used only one to be homogeneous, but I can changed it if it's really necessary.

Comment (ldez) [inline integration/acme_test.go]: they provide data shared between test (from the setup), not for tests data.

**Pull request merged**